pkg/api/v1beta1/conversion.go

```diff
@@ -1113,6 +1114,48 @@ func init() {
 			out.TimeoutSeconds = in.TimeoutSeconds
 			return nil
 		},
+		func(in *newer.Endpoints, out *Endpoints, s conversion.Scope) error {
+			if err := s.Convert(&in.TypeMeta, &out.TypeMeta, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.ObjectMeta, &out.TypeMeta, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Protocol, &out.Protocol, 0); err != nil {
+				return err
+			}
+			for i := range in.Endpoints {
+				ep := &in.Endpoints[i]
+				out.Endpoints = append(out.Endpoints, net.JoinHostPort(ep.IP, strconv.Itoa(ep.Port)))
+			}
+			return nil
+		},
+		func(in *Endpoints, out *newer.Endpoints, s conversion.Scope) error {
+			if err := s.Convert(&in.TypeMeta, &out.TypeMeta, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.TypeMeta, &out.ObjectMeta, 0); err != nil {
+				return err
+			}
+			if err := s.Convert(&in.Protocol, &out.Protocol, 0); err != nil {
+				return err
+			}
+			for i := range in.Endpoints {
+				out.Endpoints = append(out.Endpoints, newer.Endpoint{})
+				ep := &out.Endpoints[i]
+				host, port, err := net.SplitHostPort(in.Endpoints[i])
+				if err != nil {
+					return err
+				}
+				ep.IP = host
+				pn, err := strconv.Atoi(port)
```

Comment:
Nope.  In the relentless goal of a dumber, I mean SIMPLER grammar, Go has done away with the idea of an lvalue and just has a "name" token.  ep.Port is not a name.  There's a bug open against go to allow this syntax, but it's at least a year old and not fixed.